src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx

```diff
@@ -38,12 +51,69 @@ const FaultEventMenu = ({ shapeToolData, onEventUpdated, refreshTree, rootIri }:
   const [schematicDesignation, setSchematicDesignation] = useState<string | undefined>(undefined);
   const [selectedSystem] = useSelectedSystem();
 
+  const [snsOperationalFailureRate, setSnsOperationalFailureRate] = useState<number | undefined>(undefined);
+  const [snsPredictedFailureRate, setSnsPredictedFailureRate] = useState<number | undefined>(undefined);
+  const [snsManuallyDefinedFailureRate, setSnsManuallyDefinedFailureRate] = useState<number | undefined>(undefined);
+  const [externalManuallyDefinedFailureRate, setExternalManuallyDefinedFailureRate] = useState<number | undefined>(
+    undefined,
+  );
+  const [selectedRadioButton, setSelectedRadioButton] = useState<string>(RadioButtonType.Predicted);
+
+  const [snsOperationalIri, setSnsOperationalIri] = useState<string | undefined>(undefined);
+  const [snsPredictedIri, setSnsPredictedIri] = useState<string | undefined>(undefined);
+
+  const handleManuallyDefinedFailureRateChange = (event, type: ManualFailureRateType) => {
+    const inputValue = event.target.value;
+    const regex = /^[0-9]*\.?[0-9]*$/;
+    if (regex.test(inputValue)) {
+      if (type === ManualFailureRateType.Sns) {
+        setSnsManuallyDefinedFailureRate(inputValue);
+      }
+      if (type === ManualFailureRateType.External) {
+        setExternalManuallyDefinedFailureRate(inputValue);
+      }
+    }
+  };
+
+  const handleSnsBasicSelectedFailureRateChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    // TODO: Add handler for update with error
+    setSelectedRadioButton(event.target.value as RadioButtonType);
+    if (event.target.value === RadioButtonType.Predicted) {
+      // Updated when we switch to Pred. rate:
+      onEventUpdated({
+        ...shapeToolData,
+        selectedEstimate: { iri: snsPredictedIri, value: snsPredictedFailureRate },
+        probability: snsPredictedFailureRate,
+      });
+    }
+    if (event.target.value === RadioButtonType.Operational) {
+      // Updated when we switch to Oper. rate:
+      onEventUpdated({
+        ...shapeToolData,
+        selectedEstimate: { iri: snsOperationalIri, value: snsOperationalFailureRate },
+        probability: snsOperationalFailureRate,
+      });
+    }
+  };
+
+  const handleManualFailureRateUpdate = () => {
+    onEventUpdated({ ...shapeToolData, probability: snsManuallyDefinedFailureRate });
+  };
+
   const handleFailureModeClicked = (failureMode: FailureMode) => {
     setFailureModeOverview(failureMode);
     setFailureModeOverviewDialogOpen(true);
   };
 
   useEffect(() => {
+    // Clear values, after node was changed
+    setSnsPredictedFailureRate(undefined);
```

Comment:
Do we really need to do this?

Reply:
Yes. We want to be sure, that after node selection, we don't have any saved props from previous node